completion/models.py

```diff
@@ -201,6 +203,29 @@ def user_course_completion_queryset(cls, user, course_key):
         """
         return cls.objects.filter(user=user, course_key=course_key)
 
+    @classmethod
+    def latest_blocks_completed_all_courses(cls, user):
+        """
+        get all latest completions for user by course
+
+        Return value:
+            dict[courseKey]: [ modified_date, block_key ]
+        """
+
+        latest_completions_by_course = cls.objects.raw(
+            '''
+            SELECT id, course_key, max(modified) as latest
```

Comment:
@yro Instead of using the raw query here, I think you can use the django model `annotate` pretty easily to achieve the same, correct? 
What about: 
`latest_completions_by_course = cls.objects.filter(user=user).values('course_key').annotate(latest=Max('modified'))`

See https://docs.djangoproject.com/en/dev/topics/db/aggregation/#values

Reply:
@schenedx per our convo we got to here:
`        latest_completions_by_course = cls.objects.filter(user=user).values('course_key').annotate(
            modified=Max('modified')
            ).values('block_key')
`
but couldn't send the block key within the same query (eliminating the need for a secondary query). Will leave as SQL for now, and you can play with it to see if you can use the ORM.

Reply:
😭